docs/infrastructure_and_maintenance/background_tasks.md

```diff
@@ -0,0 +1,102 @@
+
+[[= product_name_base =]] Messenger supports multiple storage backends, such as Doctrine, Redis, and PostgreSQL, and gives developers the flexibility to create their own message handlers for custom use cases.
+
+## How it works
+
+[[= product_name_base =]] Messenger uses a command bus as a queue that stores messages, or commands, which tell the system what you want to happen, and separates them from the handler, which is the code that actually performs the task.
+
+The process works as follows:
+
+1. A message PHP object is dispatched, for example, `ProductPriceReindex`.
+2. The message is wrapped in an envelope, which may contain additional metadata, called stamps, for example, `DeduplicateStamp`.
+3. The message is placed in the transport queue.
+It can be a Doctrine table, a Redis queue, and so on.
+4. A worker process continuously reads messages from the queue, pulls them into the default bus `ibexa.messenger.bus` and assigns them to the right handler.
+5. A handler service processes the message (executes the command).
+You can register multiple handlers for different jobs.
+
+Here is an example of how you can extend your code and use [[= product_name_base =]] Messenger to process your tasks:
+
+### Configure package
+
+Create a config file, for example, `config/packages/ibexa_messenger.yaml` and define your transport:
+
+``` yaml
+ibexa_messenger:
+
+    # The DSN of the transport, as expected by Symfony Messenger transport factory.
+    transport_dsn:        'doctrine://default?table_name=ibexa_messenger_messages&auto_setup=false'
+    deduplication_lock_storage:
+        enabled:              true
+
+        # Doctrine DBAL primary connection or custom service
+        type:                 doctrine # One of "doctrine"; "custom"; "service"
+
+        # The service ID of a custom Lock Store, if "service" type is selected
+        service:              null
+
+        # The DSN of the lock store, if "custom" type is selected
+        dsn:                  null
+```
+
+!!! note "Supported transports"
+
+    You can define different transports: [[= product_name_base =]] Messenger has been tested to work with Redis, MySQL, PostgreSQL.
+    For more information, see [Symfony Messenger documentation](https://symfony.com/doc/current/messenger.html#transports-async-queued-messages) or [Symfony Messenger tutorial](https://symfonycasts.com/screencast/messenger/install#installing-messenger).
+
+### Start worker
+
+Use a process manager of your choice to run the following command, or make it start together with the server:
+
+``` bash
+php bin/console messenger:consume ibexa.messenger.transport --bus=ibexa.messenger.bus --siteaccess=<OPTIONAL>`
+```
+
+In multi-repository setups, the worker process always works for a SiteAccess that you indicate by using the `--siteaccess` option, therefore you may need to run multiple workers, one for each SiteAccess.
```

Comment:
Just a suggestion, what do you think about adding a link to https://doc.ibexa.co/en/latest/administration/configuration/repository_configuration/ somewhere here?